js/data/buffer.js

```diff
@@ -83,7 +83,7 @@ Buffer.prototype.setVertexAttribPointers = function(gl, program) {
  * @param gl The WebGL context
  */
 Buffer.prototype.destroy = function(gl) {
-    if (this.buffer) {
+    if (gl && this.buffer) {
```

Comment:
Implemented a (hopefully) cleaner solution in https://github.com/mapbox/mapbox-gl-js/pull/2847/commits/ff077540842941f5161d5443abf10fb76a35f5be